maestro/audit.py

```diff
@@ -361,22 +362,40 @@ def from_config(cfg):
 class ExecuteScriptAuditor(BaseAuditor):
     """Auditor that executes scripts with maestro action information."""
 
-    def __init__(self, name, script, level):
+    def __init__(self, args, script, level):
         super(ExecuteScriptAuditor, self).__init__(level)
         if not script:
             raise exceptions.InvalidAuditorConfigurationException(
                 'Missing script to execute!')
 
-        self._name = name
+        self._args = args
         self._script = script
 
+    def _format_container_dict(self, what):
+        if isinstance(what, entities.Entity):
+            return None
+        d = {}
+        for item in what:
+            x = dict(ship=item.ship.ip, service=item.service.name)
+            d[item.name]=x
+        d_str = '--hostserv={}'.format(StringIO.StringIO(json.dumps(d)).read())
```

Comment:
I still think you should pass this information via the script's stdin, otherwise you'll run into trouble if this argument is too long (command line + arguments has a fixed maximum size).